### DISCUSSION TITLE
Is there any roadmap?

### DISCUSSION THREAD

**Discussion by zhanglistar** [facebookincubator/velox#2575] [Q&A] [answered]
Thanks in advance.

**Comment by dingqiaoyi** [comment]
I'm also interested in when the GA version will be released. 
Is it likely to be released this year？@pedroerp @mbasmanova

**Reply by pedroerp** [reply]
@dingqiaoyi the library is being developed very actively, but it's unclear at this point when we will implement a proper versioning scheme.  However there are already many production deployments of Velox within Meta (and outside as reported by some collaborators).

**Comment by mbasmanova** [comment, marked as answer]
@dingqiaoyi We do not have a publicly sharable roadmap. Some of the major in-progress items are:

- decimal support
- window function support
- new optimized Parquet reader
- integration with Presto and Spark

Is there something specific you are interested in?

CC: @majetideepak

**Comment by dingqiaoyi** [comment]
> @dingqiaoyi We do not have a publicly sharable roadmap. Some of the major in-progress items are:
> 
> * decimal support
> * window function support
> * new optimized Parquet reader
> * integration with Presto and Spark
> 
> Is there something specific you are interested in?
> 
> CC: @majetideepak

Is there an expected time for Velox fully supporting TPC-H and TPC-DS ?

**Comment by mbasmanova** [comment]
> Is there an expected time for Velox fully supporting TPC-H and TPC-DS ?

TPC-H is already supported with Presto. @rui-mo Rui, do we have it supported with Spark as well?

TPC-DS will take a few more months.

**Comment by dingqiaoyi** [comment]
> > Is there an expected time for Velox fully supporting TPC-H and TPC-DS ?
> 
> TPC-H is already supported with Presto. @rui-mo Rui, do we have it supported with Spark as well?
> 
> TPC-DS will take a few more months.

Glad to see the progress ：)
Besides the decimal functions and window semantic, is there other choke-points on TPC-DS?

**Comment by rui-mo** [comment]
> > Is there an expected time for Velox fully supporting TPC-H and TPC-DS ?
> 
> TPC-H is already supported with Presto. @rui-mo Rui, do we have it supported with Spark as well?
> 
> TPC-DS will take a few more months.

Yes, we've supported TPC-H in Spark with most operators offloaded into Velox.

**Comment by mbasmanova** [comment]
> Besides the decimal functions and window semantic, is there other choke-points on TPC-DS?

@frankobe @aditi-pandit Frank, Aditi, are there any other blockers for TPC-DS?

**Reply by aditi-pandit** [reply]
TPC-DS query eval support is completed now.

**Comment by zhanglistar** [comment]
Very glad to see the progress. We are watching velox, Intel gluten and prestodb native execution project,
and very eager to use and improve performance of spark and presto now. Also we will contribute back if we have improvements. @pedroerp  @mbasmanova 
BTW, we are using clickhouse for OLAP business need, and contribute about 100 PRs now, we are bigock organization.

**Comment by pedroerp** [comment]
@zhanglistar great to hear, we can most definitely use the help! 

Both Presto and Spark integrations are quickly maturing and can already run TPC-H (and most TPC-DS). We're now iterating on performance to ensure there are no unexpected bottlenecks. 

Moving this to a github Discussion.

**Reply by iwanttobepowerful** [reply]
Is Velox fully supporting TPC-DS now?

**Reply by majetideepak** [reply]
Yes, Presto (Prestissimo) has full support for TPC-DS now.

**Reply by iwanttobepowerful** [reply]
cool!!
